tox.ini

```diff
@@ -2,8 +2,14 @@
 envlist = py27, py35, py36, py37, pypy
 
 [testenv]
+# For performance, but also for using "source" with coveragepy (https://github.com/nedbat/coveragepy/issues/268).
+usedevelop = True
 deps =
     pytest
-    pytest-cov
+    # 4.5.2a1 for py37 fix
+    coverage: https://github.com/nedbat/coveragepy/archive/58b210ad8998a9270f4ee6ff0c9054785f579b43.zip
+    coverage: pytest-cov
+setenv =
+    coverage: PYTEST_ADDOPTS=--cov --cov-report=term-missing
 commands =
     py.test {posargs}
```

Comment:
Could you add a human-oriented coverage tox env that allows someone to check coverage before pushing? Something like:

* rm .coverage and rm -r cover
* ran the tests with coverage turned on
* coverage html -d cover

Doesn't have to be in the pull request, could be a different one.

Reply:
Yes, could be useful, although `tox -e py37-coverage` should be enough and then allows you to use your preferred reporting, e.g. I would rather use `coverage report …` instead of the html report.

Reply:
..and pytest-cov does the reporting already to the terminal btw.

In general a Makefile might be better suited though, since otherwise with a separate env you  get a duplicated one:

```Makefile
cover: PY=py37
cover:
	tox -e $(PY)-coverage
	.tox/$(PY)-coverage/bin/coverage html -d $@
```
Anyway, feedback via PRs is better in general though (since it covers all envs).

Reply:
> Anyway, feedback via PRs is better in general though (since it covers all envs).

The thing I'm trying to encourage is people checking their coverage before they ever push because when someone finally makes a pull request they should think it is ready. Which tooling we have to encourage that doesn't really matter to me, just so long as such tooling is available.

Reply:
So I guess running e.g. `tox -e py37-coverage` alone would be good then?
If you want to add it to docs, you can also mention to run `coverage html -d cover` afterwards for an HTML report I guess.